### DISCUSSION TITLE
addWidget / removeWidget on index widget

### DISCUSSION THREAD

**Discussion by Spone** [algolia/instantsearch#4646] [Ideas] [answered]
Hello,

I got beaten trying to use `addWidget` on an `index`: actually [only the plural `addWidgets` exists](https://www.algolia.com/doc/api-reference/widgets/index-widget/js/#methods).

I wonder if you would be open to add the singular `addWidget`  and `removeWidget` to the index widget, in order to be on par with the [`instantsearch` supported methods](https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/#methods)?

Thanks!

**Comment by Haroenv** [comment, marked as answer]
Hi @Spone, we are planning to remove the `addWidget` & `removeWidget` from InstantSearch in the next major, which is why it wasn't added to the newer index feature.

Did you see the warning message on using addWidget that recommends using addWidgets (or do you only use the production build)? 

I'd love to also hear why you prefer addWidget(widget) over addWidgets([widget]), maybe you have a different use case than we had in mind!

Thanks!

**Reply by Spone** [reply]
OK, I was not aware of the upcoming removal of the singular forms. I did not get a warning message (I use the NPM package with Webpack).

I don't have any specific use case for `addWidget`, I just used it as a commodity when I had only one widget to add :)
I just think it's a good idea to be consistent in terms of API between `index` and `instantsearch`.
